warden/warden_daemon/tests/common/mod.rs

```diff
@@ -0,0 +1,100 @@
+use std::{
+    env,
+    fs::{remove_dir_all, remove_file},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use nix::{
+    sys::signal::{self, Signal::SIGINT},
+    unistd::Pid,
+};
+use tokio_vsock::VMADDR_CID_HOST;
+use uuid::Uuid;
+use warden_client::realm::RealmConfig;
+use warden_client::realm::{CpuConfig, KernelConfig, MemoryConfig, NetworkConfig};
+use warden_daemon::cli::Cli;
+
+pub struct PathResourceManager {
+    resource_path: Box<dyn AsRef<Path>>,
+}
+
+impl PathResourceManager {
+    pub async fn new() -> Self {
+        const TEST_FOLDER_PATH: &str = "/tmp/warden-daemon-integration-tests";
```

Comment:
Why not use https://docs.rs/mktemp/latest/mktemp/struct.Temp.html or something similar if you store the path inside the object?

Reply:
Done, thanks for the suggestion to use crate, I've choosen this one: https://crates.io/crates/tempfile.